flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/utils/DataTypeUtils.java

```diff
@@ -334,12 +334,15 @@ public static ResolvedSchema expandCompositeTypeToSchema(DataType dataType) {
         return Collections.singletonList(dataType);
     }
 
-    /** Returns the names of the flat representation in the first level of the given data type. */
+    /**
+     * Returns the names of the flat representation of the given data type. In case of {@link
+     * StructuredType}, returns the deep list of field names of the structure.
```

Comment:
the original JavaDoc was more correct: `in the first level` we don't do it deep in the sense of going through children as well

Reply:
This particular method goes deep into a structured type: https://github.com/apache/flink/blob/276e847efbfe39a567a8033dd52ff9245aee9de1/flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/utils/LogicalTypeChecks.java#L483